modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java

```diff
@@ -135,4 +145,43 @@ private String unwrapChildKey(String key)
     {
         return key.replaceAll(CHILD_ELEMENT_KEY_PREFIX, "").replaceAll(CHILD_ELEMENT_KEY_SUFFIX, "");
     }
+
+    private void checkParameterGroupExclusivenessForModel(EnrichableModel model, ResolverSet resolverSet) throws ConfigurationException
+    {
+        Optional<ParameterGroupModelProperty> parameterGroupModelProperty = model.getModelProperty(ParameterGroupModelProperty.class);
+        if (parameterGroupModelProperty.isPresent() && parameterGroupModelProperty.get().hasExclusion())
```

Comment:
Even though it is cleaner that way, I cannot throw a checked exception inside a lambda block